### PR TITLE
Feature/gcp

### PR DESCRIPTION
**Pull request by jack-ganbold**
Add GCP Feature

- Container Registry
- Cloud Run
- GKE
- GCE

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 _Cloud Container Attack Tool (CCAT) is a tool for testing security of container environments._
 
-<img src="https://raw.githubusercontent.com/RhinoSecurityLabs/ccat/master/docs/images/ccat_main_menu.png" width="1000"/>
+<img src="https://raw.githubusercontent.com/RhinoSecurityLabs/ccat/feature/gcp/docs/images/ccat_new_main_menu.png" width="1000"/>
```
Comment (SpenGietz) [inline README.md]: You can use relative URLs here so you don't have to read a picture from a different branch

Should just be able to do `src="docs/images/ccat_new_main_menu.png"`

File: README.md [hunk 2]
```diff
@@ -18,7 +18,8 @@ _Cloud Container Attack Tool (CCAT) is a tool for testing security of container
 ## Requirements
 * Python 3.5+ is required.
 * Docker is required. Note: CCAT is tested with Docker Engine 19.03.1 version.
-* AWS named profile is required.
+* Named profile is required for using AWS functionalities.
+* Service account is required for using GCP functionalities.
```
Comment (SpenGietz) [inline README.md]: A service account or access token

File: ccat.py [hunk 3]
```diff
@@ -105,15 +126,42 @@ def run_module(self, answers):
...
+            self.print_module_summary(data, gcr__pull_repos)
+
+        elif PUSH_GCR_REPOS in answers['main_menu']:
+            cli_answers = self.extentions['gcp'].ask_gcr_push_repos()
+            self.print_module_running(gcr__push_repos.module_info['name'])
+            data = gcr__push_repos.main(cli_answers)
+            self.print_module_summary(data, gcr__push_repos)
+
+        elif SWAP_SERVICE_ACCOUNT in answers['main_menu']:
+            self.extentions['gcp'].swap_service_account()
```
Comment (SpenGietz) [inline ccat.py:156]: Or  access token right?

File: ccat.py [hunk 4]
```diff
@@ -354,6 +402,234 @@ def append_configuration(self, answers):
...
+            'access_token': answers.get('access_token')
+        })
+
+        print('Successfully configured GCP\n')
+
+    def ask_gcr_enum_repos(self):
+        if self.is_configured() is False:
+            self.set_configuration()
+        
+        gcp_registries = ['gcr.io','us.gcr.io','eu.gcr.io','asia.gcr.io']
```
Comment (SpenGietz) [inline ccat.py:506]: Like you said, is there a way to get this list programmatically?

File: modules/gcr__pull_repos/main.py [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    else:
+        gcr_pull(args, data)
+
+    return data
+
+
+def summary(data):
+    out = ''
+    out += '{} GCR Repositories Pulled\n'.format(data['count']) 
+    out += 'GCR resources saved and use \'docker images\' command to check the result.\n'
```
Comment (SpenGietz) [inline modules/gcr__pull_repos/main.py]: GCR resource saved. Use the \'docker images\'  command to check the results.

File: modules/gcr__push_repos/main.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+                data['count'] = 0
+    except Exception as e:
+        print(e, file=sys.stderr)
+    
+    return data
+
+def summary(data):
+    out = ''
+    out += '{} GCR Repositories Pushed\n'.format(data['count'])
+    out += 'GCR recources saved in memory database.\n'
```
Comment (SpenGietz) [inline modules/gcr__push_repos/main.py]: resources

What database are they stored in?

File: README.md [hunk 7]
```diff
@@ -3,7 +3,7 @@
 
 _Cloud Container Attack Tool (CCAT) is a tool for testing security of container environments._
```
Comment (SpenGietz) [inline README.md:4]: Can you update the old code in setup.py and wherever else there is old code? "Container Hack Tool" and the wrong GitHub URL and things like that

**Review by SpenGietz — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jack-ganbold) [reply README.md]: That is a good way to do that but docker hub's readme can not read from this path b/c of that reason I am using githubusercontent path.

File: README.md (hunk 2 above)
Comment (jack-ganbold) [reply README.md]: Oh yeah, I forgot to update the readme after I implemented the access token feature.

File: ccat.py (hunk 3 above)
Comment (jack-ganbold) [reply ccat.py:156]: Mm, I need to change the name of that variable to SWAP_GCP_CREDS b/c we swap both service account and access token.

File: ccat.py (hunk 4 above)
Comment (jack-ganbold) [reply ccat.py:506]: I believe, there must be a way to get this programmatically. I will do that when I am possible then I will create another PR for that.

File: modules/gcr__pull_repos/main.py (hunk 5 above)
Comment (jack-ganbold) [reply modules/gcr__pull_repos/main.py]: Wonderfull, I will replace it.

File: modules/gcr__push_repos/main.py (hunk 6 above)
Comment (jack-ganbold) [reply modules/gcr__push_repos/main.py]: This is wrong info and I should fix this.

File: README.md (hunk 7 above)
Comment (jack-ganbold) [reply README.md:4]: Oh, good catch and I can definitely fix that.

Comment (jack-ganbold) [reply README.md:4]: I thought I fix that already.

File: modules/gcr__push_repos/main.py [hunk 8]
```diff
@@ -122,7 +122,7 @@ def main(args):
 def summary(data):
     out = ''
     out += '{} GCR Repositories Pushed\n'.format(data['count'])
-    out += 'GCR recources saved in memory database.\n'
+    out += 'Pushed the image. Check the Container Registery to get an info about the pushed image.\n'
```
Comment (SpenGietz) [inline modules/gcr__push_repos/main.py]: remove "an"

Should just be "to get info about the pushed image"

File: README.md [hunk 9]
```diff
@@ -18,7 +18,8 @@ _Cloud Container Attack Tool (CCAT) is a tool for testing security of container
 ## Requirements
 * Python 3.5+ is required.
 * Docker is required. Note: CCAT is tested with Docker Engine 19.03.1 version.
-* AWS named profile is required.
+* Named profile is required for using AWS functionalities.
```
Comment (SpenGietz) [inline README.md]: functionalities --> functionality

File: README.md [hunk 10]
```diff
@@ -18,7 +18,8 @@ _Cloud Container Attack Tool (CCAT) is a tool for testing security of container
...
+* A service account or access token is required for using GCP functionalities.
```
Comment (SpenGietz) [inline README.md]: functionalities --> functionality

**Review by SpenGietz — CHANGES_REQUESTED** [review]

File: ccat.py [hunk 11]
```diff
@@ -354,6 +402,234 @@ def append_configuration(self, answers):
...
+        self.configuration = {
+            'service_account_json_file_path': service_account_json_file_path,
+            'access_token': access_token,
+        }
+
+        self.data = {}
+
+    def get_menu_choices_registries(self, gcp_registries=[]):
+        choices = []
+        for registery in gcp_registries:
```
Comment (SpenGietz) [inline ccat.py]: registery --> registry

Also there are 3 cases of the same thing here: https://github.com/RhinoSecurityLabs/ccat/blob/master/modules/ecr__pull_repos/main.py

File: ccat.py [hunk 12]
```diff
@@ -354,6 +402,234 @@ def append_configuration(self, answers):
...
+            choices.append({
+                'name': registery
```
Comment (SpenGietz) [inline ccat.py]: registery --> registry

File: modules/gcr__push_repos/main.py [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+                data['count'] = 0
+    except Exception as e:
+        print(e, file=sys.stderr)
+    
+    return data
+
+def summary(data):
+    out = ''
+    out += '{} GCR Repositories Pushed\n'.format(data['count'])
+    out += 'Pushed the image. Check the Container Registery to get an info about the pushed image.\n'
```
Comment (SpenGietz) [inline modules/gcr__push_repos/main.py]: registery --> registry

**Review by SpenGietz — CHANGES_REQUESTED** [review]

**Pull request merged**